### PR TITLE
[ref #586] Walkeeper: check parsing of hot standby update

### LINKED ISSUE #586: Walkeeper expects HotStandbyFeedback without checking, page server sends StandbyStatusUpdate
Motivation for #587.

Page server sending:

https://github.com/zenithdb/zenith/blob/7c62a57e543c47d656bbade775aeb964e9c6b0b7/pageserver/src/walreceiver.rs#L311

(which eventually builds the following bytes, inside of a CopyData message)

```rs
let mut buf = BytesMut::new();
buf.put_u8(STANDBY_STATUS_UPDATE_TAG);
buf.put_u64(write_lsn.into());
buf.put_u64(flush_lsn.into());
buf.put_u64(apply_lsn.into());
buf.put_i64(ts.0);
buf.put_u8(reply);

this.stream.send(buf.freeze()).await
```
([link](https://github.com/zenithdb/rust-postgres/blob/9eb0dbfbeb6a6c1b79099b9f7ae4a8c021877858/tokio-postgres/src/replication.rs#L79-L87))

Walkeeper parsing:

https://github.com/zenithdb/zenith/blob/7c62a57e543c47d656bbade775aeb964e9c6b0b7/walkeeper/src/replication.rs#L83-L86

Also worth noting that the walkeeper ends up including the tag byte in what it parses.

[Relevant postgres protocol docs](https://www.postgresql.org/docs/13/protocol-replication.html)

### PR DESCRIPTION
**Pull request by sharnoff**
[builds on #585] This fixes #586, but I'm not sure how necessary it is to actually fix this problem. With this change, we never actually call any of the `add_hs_feedback` methods, though AFAICT the actual contents of the `HotStandbyFeedback` weren't being used anyways. (And when we called `add_hs_feedback` before, it seemed to only be with bad data)

### CODE REVIEW

**Review by hlinnaka — APPROVED** [review]
Yeah, I don't think the hot-standby feedback messages are really used at the moment. The page server connects to the WAL safekeeper, but the page server never sends HotStandbyFeedback messages, only StandbyStatusUpdates. If you connected a PostgreSQL server in standby mode directly to the WAL safekeeper, it would send those HotStandbyFeedback messages.

There's fair amount a lot of code to handle the hot standby feedback info in walproposer.c, too, but I guess it's more or less dead code, too. Although if we were previously putting bogus values in the feedback info, because we were incorrectly interpreting StandbyStatusUpdate messages as HotStandbyFeedback messages, perhaps you could see those bogus values in the `pg_stat_replication` view.

File: walkeeper/src/replication.rs [hunk 1]
```diff
@@ -81,9 +85,23 @@ impl ReplicationConn {
...
         while let Some(msg) = FeMessage::read(&mut stream_in)? {
-            match msg {
+            match &msg {
                 FeMessage::CopyData(m) => {
-                    let feedback = HotStandbyFeedback::des(&m)?;
+                    // There's two possible data messages that the client is supposed to send here:
+                    // `HotStandbyFeedback` and `StandbyStatusUpdate`. We only handle hot standby
+                    // feedback.
+
+                    let tag_byte = m.first();
```
[the remark this answers is not in the document]
Comment (sharnoff) [reply walkeeper/src/replication.rs]: Fixed :)

**Comment by arssher** [comment]
Sending garbage like currently isn't particularly good, but 'unexpected message' in logs embarasses as well. I'd suggest silently dropping the message.

**Comment by sharnoff** [comment]
@arssher Currently we aren't actually printing 'unexpected message' normally.  That only gets triggered for things that are outside of the normal replication protocol communications.

**Comment by arssher** [comment]
Sorry, my eye skipped some symbols, never mind. There is also a typo in `STANDBY_STATUS_UPDTATE_TAG` name.

**Review by arssher — APPROVED** [review]

**Pull request merged**